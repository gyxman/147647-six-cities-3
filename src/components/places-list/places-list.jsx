import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {PlaceCard} from "../place-card/place-card.jsx";

export class PlacesList extends PureComponent {

  constructor(props) {
    super(props);

    this.state = {
      activePlace: {},
    };
  }

  render() {
    const {offers} = this.props;

    return (<div className="cities__places-list places__list tabs__content">
      {offers.map((offer, i) => (<PlaceCard key={`${i}-${offer.name}`} offer={offer} onHover={this._hoverHandler.bind(this)} />))}
    </div>
    );
  }

  _hoverHandler(place) {
    this.setState({
      activePlace: place
    });
  }
}

PlacesList.propTypes = {
  offers: PropTypes.array.isRequired,
};
