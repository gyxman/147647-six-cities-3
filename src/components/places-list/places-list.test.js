import React from "react";
import renderer from "react-test-renderer";
import {PlacesList} from "./places-list.jsx";

it(`Если приложение загрузилось, то компонент PlacesList отрисовался`, () => {
  const tree = renderer
    .create(<PlacesList
      offers={[{}]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
