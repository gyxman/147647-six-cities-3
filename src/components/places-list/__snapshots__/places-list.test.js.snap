// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Если приложение загрузилось, то компонент PlacesList отрисовался 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
          </b>
          <span
            className="place-card__price-text"
          >
            / 
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
        />
      </h2>
      <p
        className="place-card__type"
      />
    </div>
  </article>
</div>
`;
